esx_service/cli/vmdkops_admin_test.py

```diff
@@ -228,6 +229,85 @@ def test_ls_no_args(self):
         for i in range(len(volumes)):
             self.assertEqual(volumes[i]['filename'], rows[i][0] + '.vmdk')
 
+class TestSet(unittest.TestCase):
+    """ Test set functionality """
+
+    def setUp(self):
+        """ Setup run before each test """
+        self.vol_count = 0
+        self.cleanup()
+        for (datastore, url_name, path) in vmdk_utils.get_datastores():
+            if not self.mkdir(path):
+                continue
+            for id in range(5):
+                volName = 'testvol' + str(id)
+                fullpath = os.path.join(path, volName + '.vmdk')
+                self.assertEqual(None,
+                                 vmdk_ops.createVMDK(vm_name='test-vm',
+                                                     vmdk_path=fullpath,
+                                                     vol_name=volName))
+                self.vol_count += 1
+
+    def tearDown(self):
+        """ Cleanup after each test """
+        self.cleanup()
+
+    def mkdir(self, path):
+        """ Create a directory if it doesn't exist. Returns pathname or None. """
+        if not os.path.isdir(path):
+            try:
+                os.mkdir(path)
+            except OSError as e:
+                return None
+        return path
+
+    def cleanup(self):
+        for v in self.get_testvols():
+            self.assertEqual(
+                None,
+                vmdk_ops.removeVMDK(os.path.join(v['path'], v['filename'])))
+
+    def get_testvols(self):
+        return [x
+                for x in vmdk_utils.get_volumes()
+                if x['filename'].startswith('testvol')]
+
+    def test_set_access(self):
+        volumes = self.get_testvols()
+        self.assertEqual(len(volumes), self.vol_count)
+        for v in volumes:
+            attach_as_opt = random.choice(kv.ATTACH_AS_TYPES)
+            vol_arg = '@'.join([v['filename'].replace('.vmdk', ''), v['datastore']])
```

Comment:
nit: please add a comment about intended result of this operation (i.e. "generates string like '...' ") , it is hard to parse if the line is doing what it intents to do s the intent is not clear.